docs/dev/tutorial/android-debugging.md

```diff
@@ -0,0 +1,127 @@
+# Prerequisites
+
+Everything required for this can be installed through the
+prereq\_android\_sdk\_install.sh script in e-mission-phone. The most important
+packages are the emulator, platform-tools, and a system-image for your
+architecture. Use `sdkmanager --list_installed` to check installed android
+packages.
+
+Add `$ANDROID_HOME/emulator` and `$ANDROID_HOME/platform-tools` to your `$PATH`
+if they aren't already added.
+
+# Setting Up the Target
+
+Setup the android emulator or connect a phone over usb/wifi.
+
+## Android Emulator
+
+Note that the android emulator does not simulate the geofence so trips need to
+be started manually in the developer options.
+
+Create an Android Virtual Device:
+
+```
+avdmanager avd create \
+-k 'system-images;android-34;google_apis_playstore;x86_64' \
+-n <avd-name>
+```
+
+The default hardware profile can be used.
+
+Make sure to replace x86_64 if you're on a different host architecture. Use
+`sdkmanager --list` to see available architectures to install.
+
+Start the emulator:
+
+```
+emulator -avd <avd-name>
+```
+
+`adb` should connect to the emulator automatically, run `adb devices` to verify.
+
+## Real Phone
+
+Turn on developer options and enable USB debugging then plug the phone into your
+computer.
+
+# Loading the Test Application
+
+Once `adb` is connected to the test phone you can use `adb install <path to apk>`
+to install the test application.
+
+# Java Debugger
+
+Make sure developer options are enable (repeatedly tap the build number in
+settings). Go to "Select debug app" in developer options and choose the test
+app. This will start a jdwp port any time the app is launched. Run `adb jdwp` to
+get the active jdwp port.
```

Comment:
Interesting! I don't remember having to do this, but I was on an older version of android.